torchdet3d/builders/model_builder.py

```diff
@@ -0,0 +1,149 @@
+                                    num_classes=cfg.model.num_classes, export_mode=export_mode, **params)
+            if cfg.model.pretrained and not export_mode:
+                init_pretrained_weights(model, key='mobilenetv3_large')
+
+    elif cfg.model.name == 'mobilenetv3_small':
+            params = model_params['mobilenetv3_small']
+            model = model_wraper(model_class=MobileNetV3, output_channels=1024,
+                                    num_classes=cfg.model.num_classes, export_mode=export_mode, **params)
+            if cfg.model.pretrained and not export_mode:
+                init_pretrained_weights(model, key='mobilenetv3_small')
+
+    return model
+
+def model_wraper(model_class, output_channels, num_points=18,
+                    num_classes=1, pooling_mode='avg', export_mode=False, **kwargs):
+    class ModelWrapper(model_class):
+        def __init__(self, output_channel=output_channels, **kwargs):
+            super().__init__(**kwargs)
+            self.regressors = nn.ModuleList()
+            for trg_id, trg_num_classes in enumerate(range(num_classes)):
+                self.regressors.append(self._init_regressors(output_channel))
+            self.classifier = nn.Sequential(
+                nn.Dropout(0.5),
+                nn.Linear(output_channel, num_classes),
+            )
+
+            self.sigmoid = nn.Sigmoid()
+
+        @staticmethod
+        def _init_regressors(output_channel):
+            return nn.Linear(output_channel, num_points)
+
+        @staticmethod
+        def _glob_feature_vector(x, mode, reduce_dims=True):
+            if mode == 'avg':
+                out = F.adaptive_avg_pool2d(x, 1)
+            elif mode == 'max':
+                out = F.adaptive_max_pool2d(x, 1)
+            elif mode == 'avg+max':
+                avg_pool = F.adaptive_avg_pool2d(x, 1)
+                max_pool = F.adaptive_max_pool2d(x, 1)
+                out = avg_pool + max_pool
+            else:
+                raise ValueError(f'Unknown pooling mode: {mode}')
+
+            if reduce_dims:
+                return out.view(x.size(0), -1)
+            return out
+
+        def forward_to_onnx(self, x):
+            ''' use trained classificator to predict object's class and
+                choose according head for this '''
+            features = self.extract_features(x)
+            pooled_features = self._glob_feature_vector(features, mode=pooling_mode)
+            if num_classes > 1:
+                targets = self.classifier(pooled_features)
+            else:
+                targets = torch.zeros((x.size(0)), dtype=torch.float32)
+
+            predicted_output = self.regressors[0](pooled_features).view(1, x.size(0), num_points // 2, 2)
+            if len(self.regressors) > 1:
+                for id, reg in enumerate(self.regressors[1:]):
+                    predicted_output = torch.cat((predicted_output, reg(pooled_features).view(1, x.size(0), num_points // 2, 2)))
```

Comment:
This approach to concatenation could slow down inference in IE. I'd rather put the outputs of all regressors to a list and then called `torch.cat` on it.